### PR TITLE
Create separate folder per module containing scripts to be executed on import

### PR DESCRIPTION
**Pull request by maddieclayton**
## Description
https://github.com/Azure/azure-powershell/issues/4855

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by cormacpayne** [comment]
@maddieclayton a few modules currently have `<SERVICE>Startup.ps1` scripts that get ran when their module is imported:

- `AnalysisServices`
- `Cdn`
- `HDInsight`
- `Network`
- `OperationalInsights`
- `Profile`
- `RecoveryServices.SiteRecovery`
- `RecoveryServices`
- `Sql`
- `Websites`

These modules have a class that implements `IModuleAssemblyInitializer`, which overrides the `OnImport()` method and runs the startup script. I think we should move the startup script to the corresponding `StartupScripts` folder and remove the `IModuleAssemblyInitializer` inheritance where we can (it looks like this should be everywhere but `Profile`). Then we would just need to verify the scripts ran after a successful packaging job.

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands.AnalysisServices.csproj [hunk 1]
```diff
@@ -78,6 +81,7 @@
       <SubType>Designer</SubType>
     </EmbeddedResource>
   </ItemGroup>
+  <ItemGroup />
```
Comment (markcowl) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands.AnalysisServices.csproj]: Remove the extra ItemGroup

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands.AnalysisServices.csproj [hunk 2]
```diff
@@ -70,6 +70,9 @@
     <None Include="packages.config">
       <SubType>Designer</SubType>
     </None>
+    <None Include="StartupScripts\*.ps1">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
```
Comment (markcowl) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices/Commands.AnalysisServices.csproj]: Mor module that don't yet have this, I would comment this out so we don't have an unnecessary dot-include in the psm1.

**Comment by cormacpayne** [comment]
@azuresdkci test this please

**Comment by cormacpayne** [comment]
@maddieclayton you'll need to regenerate the `.wxi` file since you are removing the startup scripts that were packaged into the installer

**Comment by markcowl** [comment]
on demand run here:https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/234/ 
signign run here:https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-sign/29/

@maddieclayton  Need to verify a sample of aliases from each module before merging

**Comment by maddieclayton** [comment]
sign run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-sign/37/

**Comment by maddieclayton** [comment]
signed job here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-sign/39/

**Comment by maddieclayton** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/254/

**Comment by maddieclayton** [comment]
on demand here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/257/
sign job here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-sign/40/

**Comment by maddieclayton** [comment]
@markcowl can you take a quick look? I've changed a fair amount since you've approved.

**Pull request merged**